csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -2273,6 +2280,21 @@ private predicate recordProperty(RecordType t, ContentSet c, string name) {
   )
 }
 
+/**
+ * Holds if data can flow from `node1` to `node2` via an assignment to
+ * the content set `c` of a delegate call.
+ *
+ * If there is a delegate call f(x), then we store "x" on "f"
+ * using a delegate argument content set.
+ */
+private predicate storeStepDelegateCall(Node node1, ContentSet c, Node node2) {
```

Comment:
The old implementation had the issue that it could potentially mix CFG nodes belonging to different CFG splits. Moreover, using `DelegateLikeCall` instead of just `DelegateCall` means that function pointers are also taken into account.